cli/crates/production-server/src/config.rs

```diff
@@ -107,18 +107,26 @@ pub struct TlsConfig {
     pub key: PathBuf,
 }
 
-#[derive(Debug, PartialEq, serde::Deserialize, Default)]
+#[derive(Debug, serde::Deserialize, Default, Clone)]
 #[serde(deny_unknown_fields)]
 pub struct TrustedDocumentsConfig {
     /// If true, the engine will only accept trusted document queries. Default: false.
     #[serde(default)]
     pub enabled: bool,
-    /// Optional name of the header that can be set to bypass trusted documents enforcement, when `enabled = true`. Only meaningful in combination with `bypass_header_value`.
+    /// See [BypassHeader]
+    #[serde(flatten)]
+    pub bypass_header: BypassHeader,
```

Comment:
bypass_header should be `Option` and `bypass_header_name` shouldn't, no?

Reply:
That's where I started but there are two downsides to that:

- small one: errors are worse (it's still the case with the current solution)
- big one: since `flatten` will swallow the error and put a `None` on the field if deserialization fails, you don't get an error when the env var is missing in a situation like ` bypass_header_value = "secret-{{ env.TEST_HEADER_SECRET }}" ` (there's a test for it).

Reply:
ok, fair points 👍